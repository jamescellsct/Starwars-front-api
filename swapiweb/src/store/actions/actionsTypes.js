//Peoples Actions Types
const FETCH_PEOPLE_STARTED = "FETCH_PEOPLE_STARTED";
const FETCH_PEOPLE_SUCCEEDED = "FETCH_PEOPLE_SUCCEEDED";
const FETCH_PEOPLE_FAILED = "FETCH_PEOPLE_FAILED";
const UPDATE_PEOPLE_IS_FETCHING = "UPDATE_PEOPLE_IS_FETCHING";
const UPDATE_PEOPLE_SUCCEEDED = "UPDATE_PEOPLE_SUCCEEDED";
const UPDATE_PEOPLE_FAILED = "UPDATE_PEOPLE_FAILED";
export const peoplesAction = {
  FETCH_PEOPLE_STARTED,
  FETCH_PEOPLE_SUCCEEDED,
  FETCH_PEOPLE_FAILED,
  UPDATE_PEOPLE_IS_FETCHING,
  UPDATE_PEOPLE_SUCCEEDED,
  UPDATE_PEOPLE_FAILED
};

//Planets Actions Types
const FETCH_PLANETS_STARTED = "FETCH_PLANETS_STARTED";
const FETCH_PLANETS_SUCCEEDED = "FETCH_PLANETS_SUCCEEDED";
const FETCH_PLANETS_FAILED = "FETCH_PLANETS_FAILED";
const UPDATE_PLANETS_IS_FETCHING = "UPDATE_PLANETS_IS_FETCHING";
const UPDATE_PLANETS_SUCCEEDED = "UPDATE_PLANETS_SUCCEEDED";
const UPDATE_PLANETS_FAILED = "UPDATE_PLANETS_FAILED";
export const planetsAction = {
  FETCH_PLANETS_STARTED,
  FETCH_PLANETS_SUCCEEDED,
  FETCH_PLANETS_FAILED,
  UPDATE_PLANETS_IS_FETCHING,
  UPDATE_PLANETS_SUCCEEDED,
  UPDATE_PLANETS_FAILED
};

//Starships Actions Types
const FETCH_STARSHIPS_STARTED = "FETCH_STARSHIPS_STARTED";
const FETCH_STARSHIPS_SUCCEEDED = "FETCH_STARSHIPS_SUCCEEDED";
const FETCH_STARSHIPS_FAILED = "FETCH_STARSHIPS_FAILED";
const UPDATE_STARSHIPS_IS_FETCHING = "UPDATE_STARSHIPS_IS_FETCHING";
const UPDATE_STARSHIPS_SUCCEEDED = "UPDATE_STARSHIPS_SUCCEEDED";
const UPDATE_STARSHIPS_FAILED = "UPDATE_STARSHIPS_FAILED";
export const starshipsAction = {
  FETCH_STARSHIPS_STARTED,
  FETCH_STARSHIPS_SUCCEEDED,
  FETCH_STARSHIPS_FAILED,
  UPDATE_STARSHIPS_IS_FETCHING,
  UPDATE_STARSHIPS_SUCCEEDED,
  UPDATE_STARSHIPS_FAILED
};

//Vehicles Actions Types
const FETCH_VEHICLES_STARTED = "FETCH_VEHICLES_STARTED";
const FETCH_VEHICLES_SUCCEEDED = "FETCH_VEHICLES_SUCCEEDED";
const FETCH_VEHICLES_FAILED = "FETCH_VEHICLES_FAILED";
const UPDATE_VEHICLES_IS_FETCHING = "UPDATE_VEHICLES_IS_FETCHING";
const UPDATE_VEHICLES_SUCCEEDED = "UPDATE_VEHICLES_SUCCEEDED";
const UPDATE_VEHICLES_FAILED = "UPDATE_VEHICLES_FAILED";
export const vehiclesAction = {
  FETCH_VEHICLES_STARTED,
  FETCH_VEHICLES_SUCCEEDED,
  FETCH_VEHICLES_FAILED,
  UPDATE_VEHICLES_IS_FETCHING,
  UPDATE_VEHICLES_SUCCEEDED,
  UPDATE_VEHICLES_FAILED
};

//Pagination Action Types
const PAGE_SELECTED = "PAGE_SELECTED";
export const paginationAction = {
  PAGE_SELECTED
};
