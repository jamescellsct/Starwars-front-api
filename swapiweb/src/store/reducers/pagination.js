export default function pagination(state = {}, action) {
  return state;
}
