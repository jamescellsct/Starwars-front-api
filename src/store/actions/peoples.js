import { peoplesAction } from "./actionsTypes";
import { getPeople } from "../../services/api/peopleService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchPeopleStarted = () => ({
  type: peoplesAction.FETCH_PEOPLE_STARTED
});

const fetchPeopleSucceeded = people => ({
  type: peoplesAction.FETCH_PEOPLE_SUCCEEDED,
  people
});

const fetchPeopleFailed = error => ({
  type: peoplesAction.FETCH_PEOPLE_FAILED,
  error
});

export const fetchPeople = () => dispatch => {
  dispatch(fetchPeopleStarted());

  return getPeople()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchPeopleSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchPeopleFailed(error));
    });
};

const updatePeopleIsFetching = () => ({
  type: peoplesAction.UPDATE_PEOPLE_IS_FETCHING
});

const updatePeopleSucceeded = people => ({
  type: peoplesAction.UPDATE_PEOPLE_SUCCEEDED,
  people
});

const updatePeopleFailed = error => ({
  type: peoplesAction.UPDATE_PEOPLE_FAILED,
  error
});

export const updatePeople = page => dispatch => {
  dispatch(updatePeopleIsFetching());

  getPeople(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updatePeopleSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updatePeopleFailed(error));
    });
};
