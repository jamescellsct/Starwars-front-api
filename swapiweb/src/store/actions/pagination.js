import { paginationAction } from "./actionsTypes";

const pageSelected = pagination => ({
  type: paginationAction.PAGE_SELECTED,
  pagination
});

export const selectPagePagination = page => dispatch => {
  dispatch(
    pageSelected({
      page
    })
  );
};
