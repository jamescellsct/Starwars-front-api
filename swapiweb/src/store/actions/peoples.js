import { peoplesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchPeopleStarted = () => ({
  type: peoplesAction.FETCH_PEOPLE_STARTED
});

const fetchPeopleSucceeded = people => ({
  type: peoplesAction.FETCH_PEOPLE_SUCCEEDED,
  people
});

const fetchPeopleFailed = error => ({
  type: peoplesAction.FETCH_PEOPLE_FAILED,
  error
});

export const fetchPeople = () => dispatch => {
  dispatch(fetchPeopleStarted());

  swapiRequest(null, null)
    .then(response => {
      dispatch(
        fetchPeopleSucceeded({
          results: response.results,
          total: response.count
        })
      );
    })
    .catch(error => {
      dispatch(fetchPeopleFailed(error));
    });
};
