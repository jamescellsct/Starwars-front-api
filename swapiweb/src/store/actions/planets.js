import { planetsAction } from "./actionsTypes";
import { getPlanets } from "../../services/api/planetsService";
import { handleResulstWithId } from "../../utils/handleResultsWithId";

const fetchPlanetsStarted = () => ({
  type: planetsAction.FETCH_PLANETS_STARTED
});

const fetchPlanetsSucceeded = planets => ({
  type: planetsAction.FETCH_PLANETS_SUCCEEDED,
  planets
});

const fetchPlanetsFailed = error => ({
  type: planetsAction.FETCH_PLANETS_FAILED,
  error
});

export const fetchPlanets = () => dispatch => {
  dispatch(fetchPlanetsStarted());

  getPlanets()
    .then(response => {
      const { results, count } = response;
      dispatch(
        fetchPlanetsSucceeded({
          page: 1,
          results: handleResulstWithId(results, 1),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(fetchPlanetsFailed(error));
    });
};

const updatePlanetsIsFetching = () => ({
  type: planetsAction.UPDATE_PLANETS_IS_FETCHING
});

const updatePlanetsSucceeded = planets => ({
  type: planetsAction.UPDATE_PLANETS_SUCCEEDED,
  planets
});

const updatePlanetsFailed = error => ({
  type: planetsAction.UPDATE_PLANETS_FAILED,
  error
});

export const updatePlanets = page => dispatch => {
  dispatch(updatePlanetsIsFetching());
  getPlanets(page)
    .then(response => {
      const { results, count } = response;
      dispatch(
        updatePlanetsSucceeded({
          page,
          results: handleResulstWithId(results, page),
          pageCount: Math.ceil(count / 10)
        })
      );
    })
    .catch(error => {
      dispatch(updatePlanetsFailed(error));
    });
};
