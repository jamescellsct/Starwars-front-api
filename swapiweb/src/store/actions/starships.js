import { starshipsAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchStarshipsStarted = () => ({
  type: starshipsAction.FETCH_STARSHIPS_STARTED
});

const fetchStarshipsSucceeded = starships => ({
  type: starshipsAction.FETCH_STARSHIPS_SUCCEEDED,
  starships
});

const fetchStarshipsFailed = error => ({
  type: starshipsAction.FETCH_STARSHIPS_FAILED,
  error
});

export const fetchStarships = () => dispatch => {
  dispatch(fetchStarshipsStarted());

  swapiRequest(null, null, "starships")
    .then(response => {
      dispatch(
        fetchStarshipsSucceeded({
          results: response.results,
          total: response.count
        })
      );
    })
    .catch(error => {
      dispatch(fetchStarshipsFailed(error));
    });
};
