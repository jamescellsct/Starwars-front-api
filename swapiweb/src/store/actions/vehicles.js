import { vehiclesAction } from "./actionsTypes";
import { swapiRequest } from "../../services/swapi_connect";

const fetchVehiclesStarted = () => ({
  type: vehiclesAction.FETCH_VEHICLES_STARTED
});

const fetchVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.FETCH_VEHICLES_SUCCEEDED,
  vehicles
});

const fetchVehiclesFailed = error => ({
  type: vehiclesAction.FETCH_VEHICLES_FAILED,
  error
});

export const fetchVehicles = () => dispatch => {
  dispatch(fetchVehiclesStarted());

  swapiRequest(null, null, "vehicles")
    .then(response => {
      dispatch(
        fetchVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(fetchVehiclesFailed(error));
    });
};

const updateVehiclesIsFetching = () => ({
  type: vehiclesAction.UPDATE_VEHICLES_IS_FETCHING
});

const updateVehiclesSucceeded = vehicles => ({
  type: vehiclesAction.UPDATE_VEHICLES_SUCCEEDED,
  vehicles
});

const updateVehiclesFailed = error => ({
  type: vehiclesAction.UPDATE_VEHICLES_FAILED,
  error
});

export const updateVehicles = page => dispatch => {
  dispatch(updateVehiclesIsFetching());
  swapiRequest(null, page, "vehicles")
    .then(response => {
      dispatch(
        updateVehiclesSucceeded({
          results: response.results,
          page: response.page,
          pageCount: response.pageCount
        })
      );
    })
    .catch(error => {
      dispatch(updateVehiclesFailed(error));
    });
};
