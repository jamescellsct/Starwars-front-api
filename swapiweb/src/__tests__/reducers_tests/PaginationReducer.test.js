import pagination from "../../store/reducers/pagination";
import { paginationAction } from "../../store/actions/actionsTypes";

describe("Testing Pagination Reducer", () => {
  it("Should return the initial state = page: 1", () => {
    expect(pagination(undefined, {})).toEqual({
      page: 1
    });
    expect(pagination(undefined, {})).not.toBeNull();
    expect(pagination(undefined, {})).not.toBeUndefined();
    expect(pagination(undefined, {})).toMatchSnapshot();
  });

  const mockAction = {
    type: paginationAction.PAGE_SELECTED,
    pagination: {
      page: 3
    }
  };

  it("Should handle PAGE_SELECTED", () => {
    expect(pagination({}, mockAction)).toEqual({
      page: 3
    });
    expect(pagination({}, mockAction)).not.toBeNull();
    expect(pagination({}, mockAction)).not.toBeUndefined();
    expect(pagination({}, mockAction)).toMatchSnapshot();
  });
});
