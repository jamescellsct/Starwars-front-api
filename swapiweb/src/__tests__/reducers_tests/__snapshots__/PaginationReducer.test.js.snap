// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Pagination Reducer Should handle PAGE_SELECTED 1`] = `
Object {
  "page": 3,
}
`;

exports[`Testing Pagination Reducer Should return the initial state = page: 1 1`] = `
Object {
  "page": 1,
}
`;
